src/factories/calorimetry/CalorimeterClusterRecoCoG_factoryT.h

```diff
@@ -22,13 +22,18 @@ class CalorimeterClusterRecoCoG_factoryT :
         std::string tag,
         const std::vector<std::string>& input_tags,
         const std::vector<std::string>& output_tags,
-        CalorimeterClusterRecoCoGConfig cfg)
-    : JChainMultifactoryT<CalorimeterClusterRecoCoGConfig>(tag, input_tags, output_tags, cfg),
+        ConfigMap cfg)
+    : JChainMultifactoryT<ConfigMap>(tag, input_tags, output_tags, cfg),
       m_algo(tag) {
 
       DeclarePodioOutput<edm4eic::Cluster>(GetOutputTags()[0]);
       DeclarePodioOutput<edm4eic::MCRecoClusterParticleAssociation>(GetOutputTags()[1]);
 
+      // Initialize properties
+      for (const auto& [key, value] : cfg) {
+        m_algo.setProperty(key, value);
+      }
```

Comment:
This should be the other way around: loop over defined properties, look up in map, pop them from map. If map is not empty, warn or error about unused properties provided.

Reply:
The reason is that now we have to handle incorrect keys that can fail setProperty and throw. If we start from the defined keys, we don't have to worry about that; incorrect keys are just not found and left over.

Reply:
https://github.com/eic/EICrecon/pull/1154/commits/e2cce491bf6a2db5539566963ac51c7440d49d19